libs/x-content/impl/es-jackson-core/build.gradle

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+apply plugin: 'elasticsearch.build'
+apply plugin: 'com.github.johnrengelman.shadow'
+
+String jacksonVersion = "2.14.1"
+
+dependencies {
+  implementation "com.fasterxml.jackson.core:jackson-core:${jacksonVersion}"
+}
+
+['jarHell', 'thirdPartyAudit', 'forbiddenApisMain', 'splitPackagesAudit', 'checkstyleMain', 'licenseHeaders', 'spotlessJavaCheck'].each {
+  tasks.named(it).configure {
+    enabled = false
+  }
+}
+
+tasks.named("dependencyLicenses").configure {
+  mapping from: /jackson-.*/, to: 'jackson'
+}
+
+shadowJar {
```

Comment:
I believe this needs to exclude the original FilterDelegateParser class file, otherwise there will be two copies, the original and the one added by this project.

Reply:
The resulting jar only have one FilteringParserDelegate (new).
I think it wouldn't work when excluding with this snippet (similar to the hadoop approach)
```
tasks.named('shadowJar').configure {
   exclude 'com/fasterxml/jackson/core/filter/FilteringParserDelegate.class'
}
```
This results in a jar without `FilteringParserDelegate` class at all. This is because in the docs https://imperceptiblethoughts.com/shadow/configuration/filtering/
it says:

> When using exclude/include with a ShadowJar task, the resulting copy specs are applied to the final JAR contents. This means that, the configuration is applied to the individual files from both the project source set or any of the dependencies to be merged.

In hadoop api is removing the original inner classes and replacing the public class.
```
jar -tf hadoop-client-api-3.3.3.jar | grep ShutdownHookManager
org/apache/hadoop/util/ShutdownHookManager.class
org/apache/hadoop/util/ShutdownHookManager$HookEntry.class
org/apache/hadoop/util/ShutdownHookManager$1.class
org/apache/hadoop/util/ShutdownHookManager$2.class
```
The snippet has a `$` in it
```
tasks.named('shadowJar').configure {
  exclude 'org/apache/hadoop/util/ShutdownHookManager$*.class'
}
```

Reply:
yeah lets filter out the jar version of `FilteringParserDelegate` explicitly. Otherwise which version will be picked is a gradle implementation detail and could change anytime when gradle or the shadow jar plugin is updated.


We can differ between the jar version and our patched version by using the following api:

```
shadowJar {
  exclude { element ->
    element.file == null && element.path.endsWith("FilteringParserDelegate.class")
  }
  manifest {
    attributes 'Multi-Release' : 'true'
  }
}

```

Reply:
Ah, I remember now why we have that filter. It's specifically for the anonymous inner classes. So I think it's ok to just overwrite. However, please double check the jar manually to confirm there are not eg 2 entries in the zip manifest.